asset/asset.go

```diff
@@ -524,19 +524,32 @@ func NewRawKeyGenesisSigner(priv *btcec.PrivateKey) *RawKeyGenesisSigner {
 // identified by the passed key descriptor. The final tweaked public key and
 // the signature are returned.
 func (r *RawKeyGenesisSigner) SignGenesis(keyDesc keychain.KeyDescriptor,
-	gen Genesis) (*btcec.PublicKey, *schnorr.Signature, error) {
+	groupGen Genesis, newGen *Genesis) (*btcec.PublicKey,
+	*schnorr.Signature, error) {
 
 	if !keyDesc.PubKey.IsEqual(r.privKey.PubKey()) {
 		return nil, nil, fmt.Errorf("cannot sign with key")
 	}
 
 	tweakedPrivKey := txscript.TweakTaprootPrivKey(
-		*r.privKey, gen.GroupKeyTweak(),
+		*r.privKey, groupGen.GroupKeyTweak(),
 	)
 
+	// The key tweak and signed genesis are the same when first using
+	// a group key. For future use, the asset type of the new asset must
+	// match the first asset of the group.
+	if newGen == nil {
+		newGen = &groupGen
+	} else if groupGen.Type != newGen.Type {
```

Comment:
FWIW, the msg value here we use already includes the asset type, which means that you'll always get a unique asset group key for assets w/ all the same values, but a diff asset type.

Reply:
Ah true, didn't realize that - So sounds like we don't need to check the asset type when signing at all.

I was thinking of the case where we pass a mismatched group key, that was local but for a mismatched asset type. This check probably belongs at a higher level.

Reply:
From offline discussion: We still need this check, but can remove it from `GenesisSigner.SignGenesis()` since it is already being performed at an earlier point, when a seedling group key is being validated.

https://github.com/lightninglabs/taro/blob/a770c7b94050548af8ac4b0ff5b5870a1a34e3d0/tarogarden/seedling.go#L159